### PR TITLE
Implement docker exec healthchecks.

### PR DESCRIPTION
**Pull request by mbruggmann**

### CODE REVIEW

**Comment by mbruggmann** [comment]
@martnu @rohansingh @davidxia

**Comment by martnu** [comment]
Looks good to me! :+1:

**Comment by davidxia** [comment]
@mbruggmann Hey, I'll be able to look at this soon. Been pretty busy lately.

File: helios-client/src/main/java/com/spotify/helios/common/JobValidator.java [hunk 1]
```diff
@@ -384,7 +385,8 @@ private boolean validateRepositoryName(final String repositoryName,
     final Set<String> errors = Sets.newHashSet();
 
     if (healthCheck instanceof ExecHealthCheck) {
-      if (isNullOrEmpty(((ExecHealthCheck) healthCheck).getCommand())) {
+      List<String> command = ((ExecHealthCheck) healthCheck).getCommand();
```
Comment (rohansingh) [inline helios-client/src/main/java/com/spotify/helios/common/JobValidator.java]: `final List<String> command`

File: helios-system-tests/src/main/java/com/spotify/helios/system/SystemTestBase.java [hunk 2]
```diff
@@ -693,64 +692,13 @@ protected JobId createJob(final String name,
...
-        final HttpHealthCheck httpHealthCheck = (HttpHealthCheck) job.getHealthCheck();
-        args.add("--http-check=" + httpHealthCheck.getPort() + ":" + httpHealthCheck.getPath());
-      } else if (job.getHealthCheck() instanceof TcpHealthCheck) {
-        args.add("--tcp-check=" + ((TcpHealthCheck) job.getHealthCheck()).getPort());
-      }
-    }
-
-    args.add("--");
-    args.addAll(job.getCommand());
+    final String serializedConfig = Json.asNormalizedString(job);
```
Comment (rohansingh) [inline helios-system-tests/src/main/java/com/spotify/helios/system/SystemTestBase.java]: Great idea.

**Comment by rohansingh** [comment]
:+1:

**Comment by davidxia** [comment]
@rohansingh I think @mbruggmann is on vacation. I made some minor edits here https://github.com/mbruggmann/helios/pull/1/files. Not sure when he's back.

**Comment by mbruggmann** [comment]
I'm back may 26 but feel free to edit/improve/merge/discard this as you please.

**Comment by davidxia** [comment]
@rohansingh I squashed into 1 commit here https://github.com/spotify/helios/pull/482

**Comment by davidxia** [comment]
superseded by https://github.com/spotify/helios/pull/482

**Pull request closed without merging**